src/server/node_manager.c

```diff
@@ -1133,11 +1118,11 @@ shallow_vnode_dup(struct pbsnode *vnode)
 	/*
 	 * Copy vnode elements (same order as "struct pbsnode" element definition)
 	 */
-	if (vnode_dup->nd_moms) {
-		/* free the initialize_pbsnode() allocation before assigning */
-		free(vnode_dup->nd_moms);
-		vnode_dup->nd_moms = NULL;
-	}
+	/* free the initialize_pbsnode() allocation before assigning */
+
+	free(vnode->nd_name);
```

Comment:
just trying to understand, why do we need to free 'vnode' before duping it? and why are we only freeing nd_name?